### DISCUSSION TITLE
How to pass parameter value to SQLAlchemy hybrid_method

### DISCUSSION THREAD

**Discussion by denisvitez** [sqlalchemy/sqlalchemy#9397] [Usage Questions]
Hello,

I am wondering if it's possible to pass a value for parameter to SQLAlchemy BaseQuery when performing a select.

So let's say I have a price calculation that is dependent on the selected customer for the query. I wanted to use a hybrid_property that would receive this customer ID and for each item, it would then calculate the actual price.

So let's say I have the following models:

    Item(Model):
        __tablename__ = 'Owners'
        Id = Column(Integer, primary_key=True, autoincrement=True)
        Name = Column(String(100))
        Category = Column(String(10))
        Width = Column(Integer)
        BasePrice = Column(Float)
    
    Discount(Model):
        __tablename__ = 'Vehicles'
        Id = Column(Integer, primary_key=True, autoincrement=True)
        CustomerCode = Column(String(20))
        Category = Column(String(10))
        Discount = Column(Float)

In order to calculate the actual price I would then write a `hybrid_method` on the Item class:

```
@hybrid_method
def actual_price(self, customer_code):
    price_exception = Discount.query.filter_by(CustomerCode=customer_code, Category=self.Category).first()
    return self.BasePrice * (100 - price_exception.Discount)
```

I have no idea how to pass this customer_code to my query when performing a selection:
`Item.query.filter(Item.Width >= 10).all()`

So that the results would already have the actual price calculated (if this is even possible).

**Comment by CaselIT** [comment]
Hi,

I'm not sure if that's supported. The example only shows using it on a condition, so I don't think there is explicit support for having `actual_price` returned by a select

@zzzeek do you have suggestions here?

**Reply by zzzeek** [reply]
the example shows them running a query inside of hybrid_method, which you can do if you really wanted to, though it wouldn't be considered a great pattern.

it looks like they want to multiply out by a scalar subquery.   they can do that too, they just wouldn't call first() on the inner query like that, they'd use scalar_subquery().

then they'd query like `query(Item).filter(Item.actual_price(code) == <whatever>)` .   Which seems obvious so I am likely not understanding what is actually meant here.

**Reply by CaselIT** [reply]
I think the question is to return that expression in the select list

**Reply by zzzeek** [reply]
`select(Item.actual_price(code))`  ?

**Reply by CaselIT** [reply]
but I guess it does not get loaded into the instance. That would make sense since in the instance `actual_price` is the method

**Reply by zzzeek** [reply]
we have query_expression for that
